File: 卡码/代码随想录/1/04-字符串/Go/04-左旋转字符串/LeetCode_0182.go
//LCR 182. 动态口令
package main

func dynamicPassword(password string, target int) string {
	n:=len(password)
	chs:=[]byte(password)
	reverse(chs,0,target-1)
	reverse(chs,target,n-1)
	reverse(chs,0,n-1)
	return string(chs)
}

func reverse(chs []byte,start,end int){
	for start<end {
		chs[start],chs[end]=chs[end],chs[start]
		start++
		end--
	}
}

File: 卡码/代码随想录/1/04-字符串/Js/04-左旋转字符串/LeetCode_0182.js
//LCR 182. 动态口令
const dynamicPassword = function(password, target) {
  let n=password.length;
  let arr=Array.from(password);
  reverse(arr,0,target-1);
  reverse(arr,target,n-1);
  reverse(arr,0,n-1);
  return arr.join("");
};
const reverse=function(str,start,end){
  while(start<end){
    [str[start],str[end]]=[str[end],str[start]];
    start++;
    end--;
  }
}